packages/date-picker/src/DatePicker/components/Calendar/Calendar.tsx

```diff
@@ -106,23 +101,43 @@ export const Calendar: React.VFC<CalendarProps> = ({
         role="dialog"
         aria-modal="true"
       >
-        <DayPicker
-          selectedDays={range ? selectedRange : value}
-          initialMonth={getInitialMonth()}
-          firstDayOfWeek={firstDayOfWeek}
-          disabledDays={disabledDays}
-          onDayClick={onDayChange}
-          navbarElement={getNavbar}
-          className={range ? calendarStyles.range : ""}
-          classNames={defaultCalendarClasses}
-          modifiers={
-            {
-              [calendarStyles.from]: modifiers?.from,
-              [calendarStyles.to]: modifiers?.to,
-            } as Modifiers
-          }
-          onKeyDown={onKeyDown && onKeyDown}
-        />
+        {mode === "single" && (
+          <DayPicker
+            mode="single"
+            selected={value}
+            defaultMonth={getdefaultMonth()}
+            weekStartsOn={weekStartsOn}
+            disabled={disabledDays}
+            onDayClick={onDayChange}
+            classNames={defaultCalendarClasses}
+            components={{
+              IconRight,
+              IconLeft,
+            }}
+          />
+        )}
+        {mode === "range" && (
+          <DayPicker
+            mode="range"
+            selected={selectedRange}
+            defaultMonth={getdefaultMonth()}
+            weekStartsOn={weekStartsOn}
+            disabled={disabledDays}
+            onDayClick={onDayChange}
+            className={calendarStyles.range}
+            classNames={defaultCalendarClasses}
+            components={{
+              IconRight,
+              IconLeft,
+            }}
+            modifiers={
+              {
+                [calendarStyles.from]: modifiers?.from,
+                [calendarStyles.to]: modifiers?.to,
+              } as DayModifiers
+            }
```

Comment:
I was going to suggest removing `as DayModifiers`, but then realised why you had it in there.

I question whether this is still required, or if there's another way to do this (`modifiers` should not be just a date range).
Either they go into DateRangePicker, or we find a different way of adding these styles.

I suggest we comment this out for the moment until we look at DateRangePicker again and properly fix it.